packages/payload/src/config/build.ts

```diff
@@ -7,7 +7,36 @@ import { sanitizeConfig } from './sanitize.js'
  * @param config Payload Config
  * @returns Built and sanitized Payload Config
  */
-export async function buildConfig(config: Config): Promise<SanitizedConfig> {
+export async function buildConfig(
+  _config: (() => Config) | Config,
+): Promise<(() => Promise<SanitizedConfig>) | SanitizedConfig> {
+  let configFn: (() => Config) | null = null
+
+  if (typeof _config !== 'function') {
+    console.warn(
+      'For optimal performance, buildConfig should be called with a function that returns a config object. Otherwise, you will notice increased memory usage and decreased performance during development.',
+    )
+    configFn = () => _config
+    // We could still return a function that returns the sanitized config here,
+    // so that it's cached properly and not loaded multiple times after every page transition.
+    // However, in order for this to be backwards compatible, we return the sanitized config directly, the old way.
+    // Otherwise, the imported config would suddenly be a function when imported, which may break standalone scripts
+    return await loadAndSanitizeConfig(configFn)
+  } else {
+    configFn = _config
+  }
+
+  if (process.env.NODE_ENV === 'production') {
+    return await loadAndSanitizeConfig(configFn)
+  } else {
+    return async () => {
+      return await loadAndSanitizeConfig(configFn)
+    }
+  }
+}
```

Comment:
Can't the performance benefits be made automatic without changing the API by using a cache?
I'm thinking something like this:
```ts
export async function buildConfig(
  _config: Config,
): Promise<SanitizedConfig> {
  // Create a wrapper function that maintains a single instance in dev
  let cachedConfig: SanitizedConfig | null = null
  let cachedConfigFn: (() => Promise<SanitizedConfig>) | null = null

  const getConfigFn = () => async (): Promise<SanitizedConfig> => {
    // In development, cache the sanitized config
    if (process.env.NODE_ENV === 'development') {
      if (!cachedConfig) {
        cachedConfig = await loadAndSanitizeConfig(() => _config)
      }
      return cachedConfig
    }
    // In production, just process normally since it's already optimized
    return await loadAndSanitizeConfig(() => _config)
  }

  // In development, return a function that returns the cached result
  if (process.env.NODE_ENV === 'development') {
    if (!cachedConfigFn) {
      cachedConfigFn = getConfigFn()
    }
    return await cachedConfigFn()
  }

  // In production, just process normally
  return await loadAndSanitizeConfig(() => _config)
}
```